daemon/cmd/daemon.go

```diff
@@ -537,6 +538,33 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		PolicyHandler:     d.policy.GetSelectorCache(),
 		DatapathHandler:   epMgr,
 	})
+	// Preallocate IDs for old CIDRs. This must be done before any Identity allocations are
+	// possible so that the old IDs are still available. That is why we do this ASAP after the
+	// new (userspace) ipcache is created above.
```

Comment:
As `kube-apiserver` is a reserved identity, it will not have the local-scope flag set, so this current restoration logic will simply ignore it when/if seen in the initial (old) ipcache dump. Depending on the order in which resources are received from k8s it may well be possible that the IP that was previously associated with the `kube-apiserver` reserved identity will be momentarily assigned a regular, locally allocated CIDR identity during the bootstrap, however. I guess the question here is if we should observe any `kube-apiserver` identities in the (old) ipcache dump and somehow add that CIDR/ID mapping into the new ipcache as well even before the relevant state is received from k8s. I'd leave that for a follow-up challenge, however.

Reply:
ref: https://github.com/cilium/cilium/blob/7d30daf013ab0dd86245bca64dd2d438d00d7dca/pkg/ipcache/metadata.go#L268

Reply:
So in this case the local identity will be restored as a regular CIDR identity (without any additional flags), so any policies on it should continue to function. Then, when additional state is received from k8s, a new identity with `reserved:kube-apiserver` label will be allocated, and policies using the kube-apiserver entity will be updated accordingly. ipcache and policy map updates do not happen at the same time, though, but on policy updates we update the policy maps first, and when that is completed, we add the new identities to ipcache. Between these events the new policy will likely drop traffic allowed to the new `kube-apiserver` identity, as the datapath still maps the traffic to the old identity.

Reply:
Makes sense, but I don't understand why this doesn't generically apply to all CIDR identities :thinking: